### PR TITLE
Add example of datastore with v2 functions.

### PR DESCRIPTION
**Pull request by JU-2094**
## Description

Add example for Cloud Functions and Firestore in Datastore Mode

## Checklist
- [X] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [X] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: functions/v2/datastore/hello-datastore/main.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Copyright 2023 Google LLC
```
Comment (rsamborski) [inline functions/v2/datastore/hello-datastore/main.py:1]:
```suggestion
# Copyright 2024 Google LLC
```

File: functions/v2/datastore/hello-datastore/main_test.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Copyright 2023 Google LLC
```
Comment (rsamborski) [inline functions/v2/datastore/hello-datastore/main_test.py:1]:
```suggestion
# Copyright 2024 Google LLC
```

**Review by rsamborski — APPROVED** [review]
Approved with minor nits.

**Review by kweinmeister — APPROVED** [review]

**Pull request merged**